client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -376,6 +420,14 @@ private StreamCut computeEndStreamCut(Stream stream, Map<Segment, Long> endSegme
                       .thenApply(v -> completeCheckpointAndFetchStreamCut(checkpointId));
     }
 
+    @Override
+    public void updateRetentionStreamCut(Map<Stream, StreamCut> streamCuts) {
+        if (getReaderGroupState().getConfigState() != ReaderGroupState.ConfigState.READY) {
+            throw new IllegalStateException("Update failed as ReaderGroup not in READY state. Retry again later.");
```

Comment:
should we opportunistically bring the readergroup to ready state and then attempt the requested work? `

Reply:
The reason we bring the readergroup to ready state is to make sure the current functionality of the APIs is not compromised. In this new API, I don't think we need to do that. Can these calls cause inconsistencies?

Reply:
no, not really, but the illegal state exception thrown from here should then be expected and handled by this api's caller. and they should retry it and not treat this as a fatal failure.

Reply:
Should I add to the javadoc then that the user should retry incase of failure?